lua/autorun/server/cfc_weapon_lockouts_init.lua

```diff
@@ -0,0 +1,40 @@
+local IsValid = IsValid
+local CurTime = CurTime
+local ACT_VM_RELOAD = ACT_VM_RELOAD
+
+local function setLockoutDuration( wep )
+    local reload = wep:SelectWeightedSequence( ACT_VM_RELOAD )
+    wep.CFC_LockoutTime = wep:SequenceDuration( reload )
+end
+
+hook.Add( "WeaponEquip", "CFC_WeaponLockouts", function( wep, ply )
+    local wepClass = wep:GetClass()
+
+    -- Sequence information and SetNextPrimaryFire are not available until the next frame
+    timer.Simple( 0, function()
+        if not IsValid( wep ) then return end
+        if not IsValid( ply ) then return end
+
+        setLockoutDuration( wep )
+
+        local nextFire = ply.WeaponLockouts[wepClass]
+        if not nextFire then return end
```

Comment:
Should you check if its already past the next fire time and remove it from the table then return here?

Reply:
true, if if's already elapsed the gun will fire twice on the next shot 👍